### PR TITLE
Guide edit 1

### PR DESCRIPTION
**Pull request by bobhy**
My first contribution, an editorial pass on the text of the guide, focusing on English grammar and usage.  I did not intend to make any technical changes, so if you find any, they are bugs in my PR!

I'm happy to work on any updates or corrections you suggest.  

Is there any write up available for the `next` keyword?  or the `|>` operator?  I see them in lots of samples, but couldn't figure them out from the compiler or library sources and don't have Mint running yet on my (Windows) PC.

### CODE REVIEW

File: reference/functions.md [hunk 1]
```diff
@@ -136,3 +151,4 @@ component Greeter {
 }
 ```
 
+You must declare at least one parameter, though you do not have to refer to it in the body, and you cannot declare the return type, but it had better be compatible with the expression context or you will hear from the compiler!
```
Comment (gdotdesign) [inline reference/functions.md]: `at least one parameter` is not a requirement (also on line 140), it's possible to write for example just `\ => void` .

File: reference/javascript-interop/decoding-objects.md [hunk 2]
```diff
@@ -63,10 +52,11 @@ component Main {
       <div><{ "Invalid JSON!" }></div>
     }
   }
-} 
+}
 ```
 
+In this case, `user` is a `Result(Object.Error, String)`.
```
Comment (gdotdesign) [inline reference/javascript-interop/decoding-objects.md]: The example is invalid, it's missing the a catch branch (it's my bad) the `user` at that point is a `User` and the `Object.Error` is handled in the catch.

**Comment by gdotdesign** [comment]
Thank you very much! :tada: I really appreciate the time you put into it.

I only found two minor things, which I can change after merging if you don't have time. 

> Is there any write up available for the `next` keyword? or the `|>` operator?

There is no documentation for them yet, I'll write something up in the coming days.

Also the guide lives in https://www.gitbook.com so If you prefer editing in a GUI instead of a text file I would be happy to add you as a writer.

**Comment by bobhy** [comment]
Fixed the niladic lambda sample, but don't trust myself on the catch branch issue above. Merged in recent edits from master.  Ready to merge the PR in?

**Comment by gdotdesign** [comment]
Yes, merged. Thank you!

**Pull request merged**